apps/meteor/app/livechat/server/hooks/markRoomResponded.ts

```diff
@@ -26,6 +27,19 @@ callbacks.add(
 			return message;
 		}
 
+		// Return YYYY-MM from moment
+		const monthYear = moment().format('YYYY-MM');
+		const isVisitorActive = await LivechatVisitors.isVisitorActiveOnPeriod(room.v._id, monthYear);
+		if (!isVisitorActive) {
+			await LivechatVisitors.markVisitorActiveForPeriod(room.v._id, monthYear);
```

Comment:
Is it better if we use the same logic as you are using with the Room? You are using `$addToSet` there... or at least just calling the update operation without checking it first (adding the query filter to the update operation, something like `markVisitorActiveForPeriodIfNotActive`), do you think this makes sense? 🤔

Reply:
I thought of having the same logic, but at the same time, it's really difficult for a room to span across multiple months... unless there's a weird workspace over there that never closes them. But it's a really low probability of happening, so the $addToSet should work there.

For visitors, they're reused infinitely, so having the check for "no more than X" it's better suited there.

The problem: on mongo 4.x, this type of update (preventing duplications) while limiting the size of the array is not possible with addToSet, so to avoid the array growing infinitely from duplicate values & limiting the size, we do the query this way